modules/react/src/deckgl.js

```diff
@@ -152,25 +164,27 @@ export default class DeckGL extends React.Component {
       // Resolve potentially relative dimensions using the deck.gl container size
       const {x, y, width, height} = viewport;
 
-      // Clone the element with width and height set per view
-      const newProps = Object.assign({}, child.props, {width, height});
-
-      // Inject map properties
-      // TODO - this is too react-map-gl specific
-      Object.assign(newProps, viewport.getMercatorParams(), {
-        visible: viewport.isMapSynched()
-      });
-
-      const clone = cloneElement(child, newProps);
+      let viewChildren = child.props.children;
+      if (typeof viewChildren === 'function') {
+        viewChildren = viewChildren({width, height, viewport, viewState});
```

Comment:
Nit: I always find code pattern gets a little hard to read, need to think twice. It can help to assign the common variable to a temp.

```
const childGeneratorFunc = viewChildren;
...
```

And/or some well placed comments.